agent/config.go

```diff
@@ -27,8 +27,17 @@ type Config struct {
 	// NotifyNodeChange channel receives new node changes from session messages.
 	NotifyNodeChange chan<- *api.Node
 
+	// NotifyTLSChange channel sends new TLS information changes, which can cause a session to restart
+	NotifyTLSChange <-chan *api.NodeTLSInfo
+
+	// NotifyRootChange channel receives new trust root certificate changes from session assignments
+	NotifyRootChange chan<- []byte
```

Comment:
(this should be in node.go - apologies, I had some of the changes partially stashed - should be pushed now)